### ISSUE TITLE
NodeBB redirects to / rather than subfolder /forum on login

### ISSUE THREAD

**Issue by joolswills** [NodeBB/NodeBB#8515]
- **NodeBB version:** 1.14.2
- **NodeBB git hash:** 28970b030d21a9e839804b0c40e0ed1c520502ac

- **Installed NodeBB Plugins:** 
	* nodebb-plugin-category-notifications (installed, enabled)
	* nodebb-plugin-composer-default (installed, enabled)
	* nodebb-plugin-dbsearch (installed, enabled)
	* nodebb-plugin-emoji (installed, disabled)
	* nodebb-plugin-emoji-android (installed, disabled)
	* nodebb-plugin-gdpr (installed, enabled)
	* nodebb-plugin-google-analytics (installed, enabled)
	* nodebb-plugin-gravatar (installed, enabled)
	* nodebb-plugin-markdown (installed, enabled)
	* nodebb-plugin-mentions (installed, enabled)
	* nodebb-plugin-ns-embed (installed, enabled)
	* nodebb-plugin-recent-cards (installed, enabled)
	* nodebb-plugin-soundpack-default (installed, enabled)
	* nodebb-plugin-spam-be-gone (installed, enabled)
	* nodebb-rewards-essentials (installed, disabled)
	* nodebb-theme-lavender (installed, disabled)
	* nodebb-theme-persona (installed, enabled)
	* nodebb-theme-slick (installed, disabled)
	* nodebb-theme-vanilla (installed, disabled)
	* nodebb-widget-essentials (installed, enabled)
- **Database type:** mongo
- **Database version:** 4.2.8

- **Exact steps to cause this issue:**
Logged in to forum at https://retropie.org.uk/forum/
- **What you expected:**
User to be redirected to https://retropie.org.uk/forum/?loggedin=true
- **What happened instead:**
After login, user is redirected to https://retropie.org.uk/?loggedin=true

Issue started recently - possibly with 1.13 or 1.14 - I did some messing about the other day and believe reverting some pathname changes to public/src/ajaxify.js helped, but I have since reset my changes. (a97e97b9c3e07491f31e5ca7849074956ed99d88  / 7e5cb72a59f79602c523c75653057beb3829a124)

my config

```
{
    "url": "https://retropie.org.uk/forum",
    "bind_address": "127.0.0.1",
    "secret": "****",
    "database": "mongo",
    "port": [4567, 4568, 4569, 4570],
    "mongo": {
        "host": "127.0.0.1",
        "port": "27017",
        "username": "nodebbUser",
        "database": "nodebb",
        "password": "****"
    },
    "redis": {
        "host":"127.0.0.1",
        "port":"6379",
        "database": 0
    }
}
```

Most of my issues with nodebb over the years have been related to installing in a subdirectory - perhaps it's not a common configuration.

**Comment by julianlam** [comment]
How long has your nodeBB been exhibiting this problem? Did it only start with the recent update?

Can you please paste the contents of your configuration file?

**Comment by joolswills** [comment]
I believe since the references commits to public/src/ajaxify.js - but I can't say exactly which version. a 1.13.x version I think.

**Comment by joolswills** [comment]
BTW the original post was updated with more information before your reply via email - including my config.It may help.

**Comment by barisusakli** [comment]
Can you share your nginx config as well?

**Comment by joolswills** [comment]
Sure.  I stripped some full path information for security.

```
server {
  listen 443 default_server ssl http2;
  listen [::]:443 default_server ssl http2;
  server_name retropie.org.uk;
  access_log [removed];
  error_log [removed] error;

  root [removed];
  index index.html index.php;

  ssl_certificate [removed];
  ssl_certificate_key [removed];

  proxy_set_header X-Real-IP $remote_addr;
  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header X-Forwarded-Proto $scheme;
  proxy_set_header Host $http_host;
  proxy_set_header X-NginX-Proxy true;

  proxy_buffers 4 256k;
  proxy_buffer_size 128k;
  proxy_busy_buffers_size 256k;
  proxy_read_timeout 180s;
  proxy_send_timeout 180s;

  proxy_redirect off;

  # Socket.IO Support
  proxy_http_version 1.1;
  proxy_set_header Upgrade $http_upgrade;
  proxy_set_header Connection "upgrade";

  location ~ ^/forum/assets/(.*) {  
    root [removed]/nodebb;
    try_files /build/public/$1 /public/$1 @nodebb;
  }

  location /forum/plugins {
    alias [removed]/nodebb/build/public/plugins;
    try_files $uri @nodebb;
  }

  location /forum {
    proxy_pass http://io_nodes;
  }

  location @nodebb {
    proxy_pass http://io_nodes;
  }
}

```

**Comment by joolswills** [comment]
Note for anyone using email - I updated the post above after editing to add a missing snippet

**Comment by barisusakli** [comment]
Take a look here https://docs.nodebb.org/configuring/proxies/nginx/#basic-with-subfolder-install

Maybe this is the issue. Might need to use

```
location /forum/ {
 // rest of config
}
```

**Comment by pitaj** [comment]
Looks like the documentation on the scaling page may be incorrect. It looks like that's where he got his config.

https://docs.nodebb.org/configuring/scaling/

**Comment by joolswills** [comment]
Yeah I think I did. Also with the changes with assets etc I reworked the config.

But I'm still unclear why it would be just the config another redirects are ok. Logout for example.

**Comment by joolswills** [comment]
I think I had issues btw before when I tried to put the assets / plugins under / forum before. I'll test it. I can also bisect the issue if you want an exact commit with my existing config. Let me know. Cheers (also updated some grammatical errors in previous post).

**Comment by joolswills** [comment]
Thanks. So the subdirectory was missing in this code path rather than it being a config issue? Do you still suggest I adjust the nginx config btw? Nginx config got complicated with the restructuring to

**Comment by barisusakli** [comment]
I would try with just this fix first.

**Comment by joolswills** [comment]
Sorry. I submitted accidentally before I finished. Thanks though.

It would be nice to have a simplified config though for nginx. I'm sure it must be possible to be able to serve up static data and pass on dynamic requests without 3 location configs.  I'll have a play with that, but perhaps it's not doable without code changes.

Thanks for the fix. Will test

**Comment by joolswills** [comment]
cherry-picked the fix into 1.14.2 and it's working great thanks.
